lib/SimpleSAML/Auth/LDAP.php

```diff
@@ -838,3 +838,57 @@ public function whoami($searchBase, $searchAttributes)
         return $dn;
     }
 }
+
+/**
+ * Set for a given DN attributes
+ *
+ * @param string $dn
+ * The DN of an element.
+ * @param array $attributes
+ * The names and value of the attribute(s) to set using ldap_modify structure;
+ * @return bool
+ * Result of operation
+ */
+public function setAttributes($dn, $attributes)
+{
+    if (is_array($attributes)) {
+        // Log each attribute write
```

Comment:
This entire block is evaluated each time, but will only result in anything when debug is enabled (which is not the common case). Would it be possible to only execure the block when debug logging is set? Or is that undesirable?